src/multiselect_contextmenu.js

```diff
@@ -0,0 +1,506 @@
+/**
+ * @license
+ * Copyright 2022 MIT
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @fileoverview Multiple selection context menu.
+ */
+
+import * as Blockly from 'blockly/core';
+import {blockSelection, hasSelectedParent} from './global';
+
+/**
+ * Modification for context menu 'Duplicate' to be available for
+ * multiple blocks.
+ */
+const registerDuplicate = function() {
+  const duplicateOption = {
+    displayText: function() {
+      let workableBlocksLength = 0;
+      blockSelection.forEach(function(id) {
+        const block = Blockly.getMainWorkspace().getBlockById(id);
+        if (duplicateOption.check(block)) {
+          workableBlocksLength++;
+        }
+      });
+      if (workableBlocksLength <= 1) {
+        return Blockly.Msg['DUPLICATE_BLOCK'];
+      } else {
+        return Blockly.Msg['DUPLICATE_BLOCK'] + ' (' +
+        workableBlocksLength + ')';
```

Comment:
I think this works in the short term but I think eventually we'd want to have alternative strings potentially given that some languages can be particular about how they pluralize given different numbers of items. For example, we could offer:

```
return Blockly.Msg['DUPLICATE_BLOCK_MULTIPLE'].replace('%1', workableBlocksLength) || Blockly.Msg['DUPLICATE_BLOCK'] + ' (' + workableBlocksLength + ')';
```

and just follow a convention that any translation X can have a version for multiples at `X_MULTIPLE`. This would give end users more control over the appearance of the final strings.

Consider applying this change across this file.

Reply:
Great! And to keep consistency with DELETE_X_BLOCKS, which is an already existing item, I've make a small change that:
```JavaScript
return Blockly.Msg['DUPLICATE_X_BLOCKS'].replace(
            '%1', workableBlocksLength) ||
            Blockly.Msg['DUPLICATE_BLOCK'] + ' (' +
            workableBlocksLength + ')';
```